src/components/expired-subscription-modal/index.jsx

```diff
@@ -2,22 +2,37 @@ import {
   useToggle, AlertModal, Button, ActionRow,
 } from '@openedx/paragon';
 import DOMPurify from 'dompurify';
-import { useSubscriptions } from '../app/data';
+import { useContext } from 'react';
+import { AppContext } from '@edx/frontend-platform/react';
+import { postUnlinkUserFromEnterprise, useEnterpriseCustomer, useSubscriptions } from '../app/data';
 
 const ExpiredSubscriptionModal = () => {
   const { data: { customerAgreement } } = useSubscriptions();
+  const {
+    authenticatedUser: { email: userEmail },
+  } = useContext(AppContext);
+  const { data: enterpriseCustomer } = useEnterpriseCustomer();
+
   const [isOpen] = useToggle(true);
   if (!customerAgreement?.hasCustomLicenseExpirationMessaging) {
     return null;
   }
+  const onClickHandler = (redirectUrl) => async (e) => {
+    e.preventDefault();
+
+    await postUnlinkUserFromEnterprise(enterpriseCustomer.uuid, userEmail);
+
+    // Redirect immediately
+    window.location.href = redirectUrl;
+  };
   return (
     <AlertModal
       title={<h3 className="mb-2">{customerAgreement.modalHeaderText}</h3>}
       isOpen={isOpen}
       isBlocking
       footerNode={(
         <ActionRow>
-          <Button href={customerAgreement.urlForButtonInModal}>
+          <Button onClick={onClickHandler(customerAgreement.urlForButtonInModal)}>
```

Comment:
Given this is now a button that performs an asynchronous action, I would recommend using `StatefulButton` ([docs](https://paragon-openedx.netlify.app/components/statefulbutton/)) to ensure we show a loading spinner in the CTA button while the `await postUnlinkUserFromEnterprise` is resolving to give users on higher latency networks feedback that an action is occurring before the redirect takes place. Otherwise, some users could be left clicking the button with no information that something is happening in the background.

Reply:
Done 👍